src/coreclr/src/jit/codegenarmarch.cpp

```diff
@@ -2477,8 +2477,10 @@ void CodeGen::genCallInstruction(GenTreeCall* call)
     {
         // Generate a direct call to a non-virtual user defined or helper method
         assert(callType == CT_HELPER || callType == CT_USER_FUNC);
+#ifdef FEATURE_READYTORUN_COMPILER
```

Comment:
No not an accident :-) All uses of `gtEntryPoint` in this file are wrapped in "#ifdef FEATURE_READYTORUN_COMPILER". That is because it is not defined when we are not building a crossgen component. It looks like someone missed this assert and it would break the build for the mergeable JIT.